cglib/src/test/java/net/sf/cglib/proxy/TestEnhancer.java

```diff
@@ -1418,6 +1418,141 @@ public Class findClass(String name) throws ClassNotFoundException {
     assertEquals(Arrays.asList(Object.class, String.class), paramTypes);
   }
 
+    public void testInterfaceBridge() throws Exception {
+        // recent versions of javac will generate a synthetic default bridge for f(Object) in B
+
+        // interface A<T> {
+        //     void f(T t);
+        // }
+        // interface B<T extends Number> extends A<T>{
+        //     void f(T t);
+        // }
+
+        final Map<String, byte[]> classes = new HashMap<String, byte[]>();
+        {
+            ClassWriter classWriter = new ClassWriter(0);
+            classWriter.visit(
+                    Opcodes.V1_8,
```

Comment:
we test w/ java1.7 too.    could you maybe add a variant of this that's based off the classes as compiled from source, and optionally keep this but bail if < java8?